src/analysis/parsing/statement.rs

```diff
@@ -496,6 +496,17 @@ impl TreeElement for HashIfContent {
                      &self.truebranch,
                      &self.elsebranch)
     }
+    fn style_check(&self, acc: &mut Vec<DMLStyleError>, rules: &CurrentRules, mut aux: AuxParams) {
+        self.evaluate_rules(acc, rules, aux);
+        self.cond.style_check(acc, rules, aux);
+
+        if let Content::Some(statement::StatementContent::Compound(_))
+            = self.truebranch.content.as_ref() {
+            aux.depth -= 1;
```

Comment:
Will this work correctly when, say, true-branch is a compound but else-branch is not?

Reply:
We did some inspection of this scenario with the dmlc compiler and it really does not allow to have one liners in #if or #else blocks.

I think this syntax requires that the braces are always present, as the compiler fails when trying to compile such scenarios (and likewise the DLS AST also throws a syntax error in those cases) so I would not worry about having blocks which are not compound (they will always be, apparently)